File: apps/api/app/models/channel.py
# why: 모듈 역할과 책임을 명확히 하기 위한 진입 주석
from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.base import TimestampMixin


class Channel(Base, TimestampMixin):
    """홈쇼핑 채널 테이블.

    - why: 편성표와 알림의 기준 단위이므로 별도 엔티티로 분리.
    """

    __tablename__ = "channels"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    channel_code: Mapped[str] = mapped_column(String(50), unique=True, index=True)
    channel_name: Mapped[str] = mapped_column(String(100))
    channel_logo_url: Mapped[str | None] = mapped_column(String(255), nullable=True)
    channel_live_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    channel_stream_url: Mapped[str | None] = mapped_column(String(500), nullable=True)

File: apps/web/src/types/channel.ts
// why: 파일 책임을 명확히 하고 유지보수를 쉽게 하기 위한 설명 주석
export interface Channel {
  id: number;
  channel_code: string;
  channel_name: string;
  channel_logo_url?: string | null;
  channel_live_url?: string | null;
  channel_stream_url?: string | null;
}
